### ISSUE TITLE
Potential conflict with IntelliJ EduTools

### ISSUE THREAD

**Issue by ahus1** [raydac/netbeans-mmd-plugin#94]
I maintain the AsciiDoc plugin for IntelliJ and a user reported an exception. I had a look and it might be a clash between EduTools and IDEA Mind Map.

The message: 
> Assertion failed: Project view pane com.igormaznitsa.ideamindmap.view.KnowledgeViewPane@4a7bd274 has the same weight as com.jetbrains.edu.learning.projectView.CourseViewPane@26af371b. Please make sure that you overload getWeight() and return a distinct weight value.

It would be great if you could have a look. Thanks!

The environment: 

IE (IntelliJ IDEA Edu) 2021.3.3
EduTools 2022.2-2021.3-864
IDEA Mind Map 1.5.0

The full exception: 

```
java.lang.Throwable: Assertion failed: Project view pane com.igormaznitsa.ideamindmap.view.KnowledgeViewPane@4a7bd274 has the same weight as com.jetbrains.edu.learning.projectView.CourseViewPane@26af371b. Please make sure that you overload getWeight() and return a distinct weight value.
    at com.intellij.openapi.diagnostic.Logger.assertTrue(Logger.java:218)
    at com.intellij.ide.projectView.impl.ProjectViewImpl.doAddPane(ProjectViewImpl.java:765)
    at com.intellij.ide.projectView.impl.ProjectViewImpl.doAddUninitializedPanes(ProjectViewImpl.java:704)
    at com.intellij.ide.projectView.impl.ProjectViewImpl.setupImpl(ProjectViewImpl.java:888)
    at com.intellij.ide.projectView.impl.ProjectViewImpl.setupImpl(ProjectViewImpl.java:862)
    at com.intellij.ide.projectView.impl.ProjectViewToolWindowFactory.createToolWindowContent(ProjectViewToolWindowFactory.java:29)
    at com.intellij.openapi.wm.impl.ToolWindowImpl.createContentIfNeeded$intellij_platform_ide_impl(ToolWindowImpl.kt:542)
    at com.intellij.openapi.wm.impl.ToolWindowImpl.getContentManager(ToolWindowImpl.kt:403)
    at com.intellij.ide.projectView.impl.ProjectViewImpl.getContentManager(ProjectViewImpl.java:1168)
    at com.intellij.ide.projectView.impl.ProjectViewImpl.changeViewCB(ProjectViewImpl.java:1212)
    at com.intellij.ide.projectView.impl.ProjectViewImpl.changeView(ProjectViewImpl.java:1185)
    at com.intellij.ide.projectView.impl.ProjectViewImpl.changeView(ProjectViewImpl.java:1180)
    at org.asciidoc.intellij.psi.AsciiDocUtil.selectFileInProjectView(AsciiDocUtil.java:1828)
    at org.asciidoc.intellij.actions.asciidoc.CreatePdfAction.lambda$actionPerformed$4(CreatePdfAction.java:107)
    at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:947)
    at org.asciidoc.intellij.actions.asciidoc.CreatePdfAction.actionPerformed(CreatePdfAction.java:104)
    at com.intellij.openapi.actionSystem.impl.ActionButton.actionPerformed(ActionButton.java:178)
    at com.intellij.openapi.actionSystem.impl.ActionButton.lambda$performAction$0(ActionButton.java:154)
    at com.intellij.openapi.actionSystem.ex.ActionUtil.performDumbAwareWithCallbacks(ActionUtil.java:265)
    at com.intellij.openapi.actionSystem.impl.ActionButton.performAction(ActionButton.java:154)
    at com.intellij.openapi.actionSystem.impl.ActionButton.processMouseEvent(ActionButton.java:449)
    at java.awt.Component.processEvent(Component.java:6419)
    at java.awt.Container.processEvent(Container.java:2263)
    at java.awt.Component.dispatchEventImpl(Component.java:5029)
    at java.awt.Container.dispatchEventImpl(Container.java:2321)
    at java.awt.Component.dispatchEvent(Component.java:4861)
    at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4918)
    at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4547)
    at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4488)
    at java.awt.Container.dispatchEventImpl(Container.java:2307)
    at java.awt.Window.dispatchEventImpl(Window.java:2790)
    at java.awt.Component.dispatchEvent(Component.java:4861)
    at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:778)
    at java.awt.EventQueue$4.run(EventQueue.java:727)
    at java.awt.EventQueue$4.run(EventQueue.java:721)
    at java.security.AccessController.doPrivileged(AccessController.java)
    at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
    at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:95)
    at java.awt.EventQueue$5.run(EventQueue.java:751)
    at java.awt.EventQueue$5.run(EventQueue.java:749)
    at java.security.AccessController.doPrivileged(AccessController.java)
    at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
    at java.awt.EventQueue.dispatchEvent(EventQueue.java:748)
    at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:891)
    at com.intellij.ide.IdeEventQueue.dispatchMouseEvent(IdeEventQueue.java:820)
    at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:757)
    at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:447)
    at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:818)
    at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:446)
    at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:805)
    at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:498)
    at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
    at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
    at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
    at java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

**Comment by raydac** [comment]
please [try the snapshot version](https://github.com/raydac/netbeans-mmd-plugin/releases/tag/1.5.1-S) of the plugin

**Comment by ahus1** [comment]
Thanks for providing this fix. 

Unfortunately the stack trace was the only thing the user reported anonymously, with no additional information on how to reproduce it. 

I tried to reproduce it after looking at the stack trace, and wasn't successful yet. I tried this with the old and the new version of the plugin. 

Looking at the code changes you added for this issue, I think it's safe to assume this is now fixed. If the issue is reported again for a newer version of the plugin, I'll comment in this issue again.
